docs/user/developing_jobs.md

```diff
@@ -6,43 +6,96 @@ Familiarity with [DiffSync](https://diffsync.readthedocs.io/en/latest/) and with
 
-   - `self.load_source_adapter`: This is mandatory to be implemented. As an example:
+```python
+from nautobot.tenancy.models import Tenant
+from nautobot_ssot.contrib import NautobotModel
+
+class TenantModel(NautobotModel):
+    """An example model of a tenant."""
+    _model = Tenant
+    _modelname = "tenant"
+    _identifiers = ("name",)
+    _attributes = ("description",)
+
+    name: str
+    description: str
+```
+
+As you can see when looking at the source code of the `nautobot.tenancy.models.Tenant` model, the fields on the model (`name` and `description`) match the names of the fields on that model one-to-one. This enables the base class `NautobotModel` to dynamically load, create, update and delete your tenants without you needing to implement this functionality yourself.
+
+For a full how-to on modelling different kinds of fields, check the corresponding [page](modelling.md) in the docs.
+
+### Creating the Nautobot Adapter
+
+Having created all your models, creating the Nautobot side adapter is very straight-forward:
+
+```python
+from nautobot_ssot.contrib import BaseAdapter
+
+from your_ssot_app.models import DiffSyncDevice, DiffSyncPrefix, DiffSyncIPAddress
+
+class NautobotAdapter(BaseAdapter):
+    top_level = ("device", "prefix")
+
+    device = DiffSyncDevice
+    prefix = DiffSyncPrefix
+    ip_address = DiffSyncIPAddress  # Not in top_level because it's a child of the prefix model
+```
+
+The `load` function is already implemented on this adapter and will automatically and recursively traverse any children relationships for you.
 
-     ```python
-     def load_source_adapter(self):
-         """Method to instantiate and load the SOURCE adapter into `self.source_adapter`."""
-         self.source_adapter = NautobotRemote(url=self.kwargs["source_url"], token=self.kwargs["source_token"], job=self)
-         self.source_adapter.load()
-     ```
+### Implementing `create`, `update` and `delete`
 
-   - `self.load_target_adapter`: This is mandatory to be implemented. As an example:
+In case you are synchronizing data _to_ and not _from_ Nautobot, you can entirely skip this step. The `nautobot_ssot.contrib.NautobotModel` class takes care of automatically.
 
-     ```python
-     def load_target_adapter(self):
-        """Method to instantiate and load the TARGET adapter into `self.target_adapter`."""
-        self.target_adapter = NautobotLocal(job=self)
-        self.target_adapter.load()
-     ```
+If you need to perform the `create`, `update` and `delete` operations on the remote system however, it will be up to you to implement those on your model.
 
-   - `self.calculate_diff`: This method is implemented by default, using the output from load_adapter methods.
+!!! note
+    You still want your models to adhere to the [modelling guide](modelling.md), because doing so gives you the auto-generated `load` function for the diffsync adapter on the Nautobot side.
```

Comment:
```suggestion
    You still want your models to adhere to the [modeling guide](modeling.md), since it provides you the auto-generated `load` function for the diffsync adapter on the Nautobot side.
```